restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -188,6 +188,10 @@ public ReactiveDocumentService(
           authorizationService.authorizeDataRead(
               authenticationSubject, namespace, collection, SourceAPI.REST);
 
+          // Don't fail this read request if the corrective DELETE statements are not authorized,
+          // simply skip DELETE batches in that case (see the Rx pipeline below).
+          boolean deleteAuthorized = db.authorizeDeleteDeadLeaves(namespace, collection);
```

Comment:
I would not have this out of the pipe and would do it only if we need to delete the leaves, otherwise we add extra latency to each call..

Reply:
good point, but can we defer that until the execution is moved to non-TPC (non-core) threads? Auth calls can block, I guess. In any case they are not guaranteed to be non-blocking, and we should never block DSE core threads.

Reply:
resolved by 0bd2d386913abe87601f1174d7ac742a57d558cc